### ISSUE TITLE
UserLocation onUpdate prop doesn't work with native render mode

### ISSUE THREAD

**Issue by Slaymd** [react-native-mapbox-gl/maps#854]
**Describe the bug**
UserLocation.onUpdate prop doesn't work with `renderMode="native"`. (same iOS / Android)

**To Reproduce**
Easy to reproduce, switch "normal" and "native" to see the difference (logged by `onLocationUpdate`). Normal mode is always sending user location and never native mode.

Example:
```js
import React from 'react';
import MapboxGL, {
    MapView,
    UserLocation
} from '@react-native-mapbox-gl/maps';

class BugReportUserLocation extends React.Component {

    onLocationUpdate = (event) => {
        console.log("On Location Update: " + event.coords.latitude + ", " + event.coords.longitude);
    }

    render() {
        return (
            <MapView style={{flex: 1}}>
                <UserLocation renderMode={'native'} visible onUpdate={this.onLocationUpdate}/>
            </MapView>
        );
    }
}

export default BugReportUserLocation
```
**Expected behavior**
User location should be logged from the first NativeUserLocation render like `renderMode="normal"`

**Versions:**
 - Platform: iOS (same bug on Android)
 - Device: iPhoneXS
 - Emulator/ Simulator: no
 - OS: 13.4.1
 - react-native-mapbox-gl Version: 8.1.0-beta.1
 - React Native Version: 0.62.2

**Comment by bdtren** [comment]
reopen!
This bug still persist on  `react-native-mapbox-gl version 8.1.0.rc9`
For now, if `renderMode="native"`: 
   - iOS: cannot trigger onUpdate
   - Android: app crash like https://github.com/react-native-mapbox-gl/maps/issues/1021#issue-697637353

**Comment by kaiykwok2** [comment]
I had the same issue, anyone could help? Thanks

**Comment by RobertSasak** [comment]
_LocationManager_ and therefore also _onUpdate_ are deliberatly disabled when _renderMode_ is set to _native_. Any reason why is it so?
```js
  needsLocationManagerRunning() {
    if (this.props.renderMode === UserLocation.RenderMode.Native) {
      return false;
    }
    return !!this.props.onUpdate || this.props.visible;
  }
```
https://github.com/react-native-mapbox-gl/maps/blob/master/javascript/components/UserLocation.js#L201

**Comment by ferdicus** [comment]
> _LocationManager_ and therefore also _onUpdate_ are deliberatly disabled when _renderMode_ is set to _native_. Any reason why is it so?
> 
> ```js
>   needsLocationManagerRunning() {
>     if (this.props.renderMode === UserLocation.RenderMode.Native) {
>       return false;
>     }
>     return !!this.props.onUpdate || this.props.visible;
>   }
> ```
> 
> https://github.com/react-native-mapbox-gl/maps/blob/master/javascript/components/UserLocation.js#L201

Hey @RobertSasak, as for _why_ this was added I'm not sure. 
The change got introduced by @mfazekas in this PR https://github.com/react-native-mapbox-gl/maps/pull/825
Then again, that was only a continuation for @mattijsf PR https://github.com/react-native-mapbox-gl/maps/pull/215

**Comment by mfazekas** [comment]
I think #1135 is a good fix for this.

**Comment by RobertSasak** [comment]
@ferdicus Thank you for your nice summary. I think it was ment to be optimalisation. In native mode we are not showing custom marker and therefor do not need to run LocationManager. upUpdate got simple forgotten.

**Comment by ferdicus** [comment]
> @ferdicus Thank you for your nice summary. I think it was ment to be optimalisation. In native mode we are not showing custom marker and therefor do not need to run LocationManager. upUpdate got simple forgotten.

Glad it helped, PR looks like a good solution 👍🏿

**Comment by ferdicus** [comment]
resolved by #1135
